core/services/gateway/handlers/vault/handler.go

```diff
@@ -437,6 +437,8 @@ func (h *handler) handleSecretsCreate(ctx context.Context, ar *activeRequest) er
 		return h.sendResponse(ctx, ar, h.errorResponse(ar.req, api.UserMessageParseError, err))
 	}
 
+	createSecretsRequest.RequestId = ar.req.ID
```

Comment:
Looks like you haven't synced the latest develop?
This code already sets requestId, and namespace correctly on the request.